### PR TITLE
[UGO-3561] Avoid sending "polling timeout" error to rollbar on all polling errors.

### PR DESCRIPTION
**Pull request by tbecerro**
Avoid sending "polling timeout" error to rollbar on all polling errors.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -36,25 +43,19 @@ const pollResource = (url: string, interval: number, timeout: number, headers: a
...
         } else if (response.ok) {
           resolve(response);
         } else {
-          if(failureCallback) {
-            failureCallback(response, {
-                name: POLLING_TIMEOUT_ERROR,
-                originalUrl,
-                timeout: timeout
-            });
-          }
```
Comment (segulino) [inline src/index.ts]: With this change we are losing the scenarios on there Utility api don't answer request, and our api throws GATEWAY_TIMEOUT error.

We mustn't remove failureCallback from here, we must validate if error is gateway timeout and (in this case) execute callback.

File: src/index.ts [hunk 2]
```diff
@@ -36,25 +43,19 @@ const pollResource = (url: string, interval: number, timeout: number, headers: a
           if (new Date().getTime() < endTime) {
             setTimeout(pollingRequest, interval, resolve, reject);
           } else {
-            if(failureCallback) {
+            if (failureCallback) {
+              console.log('polling time out');
```
Comment (segulino) [inline src/index.ts]: Don't upload your loggers on pull requests.

**Comment by tbecerro** [comment]
@segulino done :octocat:

File: src/index.ts [hunk 3]
```diff
@@ -36,25 +46,24 @@ const pollResource = (url: string, interval: number, timeout: number, headers: a
...
         } else if (response.ok) {
           resolve(response);
         } else {
-          if(failureCallback) {
+          if (existsErrorCode(errors.GATEWAY_TIMEOUT.codes, response.data))
             failureCallback(response, {
-                name: POLLING_TIMEOUT_ERROR,
-                originalUrl,
-                timeout: timeout
+              name: errors.GATEWAY_TIMEOUT.message,
```
Comment (segulino) [inline src/index.ts]: This doesn't works, our projects that consume this package are expecting another error:

image

Please analyze and test all changes.

**Pull request merged**